lib/tests/streamlit/modin_mocks.py

```diff
@@ -0,0 +1,43 @@
+# Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022-2024)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pandas as pd
+
+
+class DataFrame:
+    __module__ = "modin.pandas.dataframe"
```

Comment:
Based on an offline discussion with Lukas its enough to set the `__module__` to ensure that we react differently to the various dataframe libraries and do not need to install the actual library (so the real module has the same name). The next added methods are methods we know exist in modin; so we want to ensure that we do not accidentally delete lines in Streamlit that rely on those modules. Installing the library itself just for that would be overkill.
It does not guarantee future compatibility though as the Modin API could change and we would need integration tests for that.

@LukasMasuch could you please leave a comment along those^ lines in the code so that the *why* its done this way is captured for future readers, please?

Reply:
Added some docstring comments to all the mock classes 👍